File: code/RNA.py
class RNASolution:
    def __init__(self):
        self.dna = None
        self.readDnaSeq()

    # lendo a sequência do dna
    def readDnaSeq(self):
        with open('../input/rosalind_rna.txt', 'r') as input_data:
            self.dna = input_data.read()
            input_data.close()

    # transcrevendo a sequência de dna para rna
    def transcribingDnaToRna(self):
        rna = ""

        for base in self.dna:
            if base == "T":
                rna += "U"
                continue
            rna += base

        return rna

    # sequência rna
    def getRnaSeq(self):
        rna = self.transcribingDnaToRna()

        with open('../output/RNA.txt', 'w') as output_data:
            output_data.write(rna)
            output_data.close()


if __name__ == "__main__":
    RNASolution().getRnaSeq()

File: code/c++/RNA.cpp
#include <iostream>
#include <string>
#include <fstream>

class RNASolution
{
private:
    std::string dna_seq, rna_seq;

public:
    // lendo a sequência de DNA
    void read_dna_seq()
    {
        std::ifstream input_data("C:/Programming/Rosalind/input/rosalind_rna.txt");

        if (input_data.is_open())
        {
            std::getline(input_data, dna_seq, '\n');

            input_data.close();
        }
        else
        {
            std::cerr << "Não foi possível abrir o arquivo. " << std::endl;
        }
    }

    // transcrição da sequência de DNA para RNA
    void transcribing_dna_to_rna()
    {
        for (auto base : dna_seq)
        {
            if (base == 'T')
            {
                rna_seq += 'U';
                continue;
            }
            rna_seq += base;
        }
    }

    // escrevendo a sequência de rna
    void get_rna_seq()
    {
        read_dna_seq();
        transcribing_dna_to_rna();

        std::ofstream output_data("C:/Programming/Rosalind/output/RNA.txt");

        if (output_data.is_open())
        {
            output_data << rna_seq;

            output_data.close();
        }
        else
        {
            std::cerr << "Não foi possível abrir o arquivo. " << std::endl;
        }
    }
};

int main()
{
    RNASolution solution;
    solution.get_rna_seq();

    return 0;
}